frame/undercollateralized-loans/src/validation.rs

```diff
@@ -0,0 +1,106 @@
+use chrono::{NaiveDate, NaiveTime};
+use codec::FullCodec;
+use composable_support::validation::{TryIntoValidated, Validate};
+use composable_traits::{
+	defi::DeFiEngine,
+	oracle::Oracle as OracleTrait,
+	undercollateralized_loans::{LoanInput, MarketInput, UndercollateralizedLoans},
+};
+use frame_support::pallet_prelude::*;
+use scale_info::TypeInfo;
+use sp_runtime::traits::Zero;
+
+#[derive(Clone, Copy, RuntimeDebug, PartialEq, TypeInfo, Default)]
+pub struct CurrencyPairIsNotSame;
+
+impl<LiquidationStrategyId, Asset: Eq, BlockNumber, AccountId>
+	Validate<
+		MarketInput<LiquidationStrategyId, Asset, BlockNumber, AccountId>,
+		CurrencyPairIsNotSame,
+	> for CurrencyPairIsNotSame
+{
+	fn validate(
+		create_input: MarketInput<LiquidationStrategyId, Asset, BlockNumber, AccountId>,
+	) -> Result<MarketInput<LiquidationStrategyId, Asset, BlockNumber, AccountId>, &'static str> {
+		if create_input.currency_pair.base == create_input.currency_pair.quote {
+			Err("Base and quote currencies supposed to be different in currency pair")
```

Comment:
NIT: "Base and quote currencies supposed to be different in currency pair" -> "Base and quote currencies are supposed to be different in currency pair"

Reply:
fixed